services/mailservers/tcp_ping.go

```diff
@@ -0,0 +1,41 @@
+package mailservers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/status-im/status-go/tcpping"
+)
+
+type PingQuery struct {
+	Addresses []string `json:"addresses"`
+	TimeoutMs int      `json:"timeoutMs"`
+}
+
+type PingResult struct {
+	Address string  `json:"address"`
+	Latency *int    `json:"latency"`
+	PingErr *string `json:"error"`
+}
+
+func fromPingResultToJSON(r tcpping.Result) PingResult {
+	err := fmt.Sprint(r.PingErr)
+	return PingResult{
+		Address: r.Address,
+		Latency: &r.Latency,
+		PingErr: &err,
+	}
+}
+
+func (a *API) Ping(ctx context.Context, pq PingQuery) ([]PingResult, error) {
+	timeout := time.Duration(pq.TimeoutMs) * time.Millisecond
+	// run the checks concurrently
+	results, err := tcpping.CheckHosts(pq.Addresses, timeout)
```

Comment:
If `err` is here, I believe we can return immediately with `return nil, err`.

Reply:
done